ocaml-lsp-server/src/prefix_parser.ml

```diff
@@ -0,0 +1,31 @@
+open Re
+
+(*Regex based parser*)
+
+let name_or_label_regex =
+  Re.compile @@ Re.Posix.re {|([~?`]|let%|and%)?([a-zA-Z0-9_']|[a-zA-Z0-9_'] *\. *)+$|}
+
+(** matches let%lwt and let* style expressions. See here:https://v2.ocaml.org/manual/bindingops.html *)
+let monadic_bind =
+  Re.compile @@ Re.Posix.re {|(let|and)([$&*+\-/=>@^|.]|(%[a-zA-Z0-9_']*))$|}
```

Comment:
Can you use the combinators from `Re` instead of `Re.Posix` please.

Reply:
How set are you on this? I far far prefer to use standard regex because 
a) it's something universal
b) it means it's easy to test and debug using tools like https://regexr.com/ 
But if you think there are good reasons to use the combinators I can convert it

Reply:
I think the main issue with the way you've written it is that you're using capturing groups that you don't actually need. That's a little confusing and also slows down the matching.

Reply:
Also, I think that if you used the combinators, constructing the "reverse" regex would be much simpler.